Arriba/Arriba.Test/CustomColumn.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using Arriba.Model;
+using System.Collections.Generic;
+using Arriba.Model.Expressions;
+using Arriba.Structures;
+using Arriba.Serialization;
+
+namespace Arriba.Test
+{
+    public static class CustomColumnSupport
+    {
+        public static void RegisterCustomColumns()
+        {
+            ColumnFactory.AddColumnCreator("color", (details, columnComponents, initialCapacity) =>
+            {
+                IColumn<ColorColumn.ComparableColor> coreColumn = new ColorColumn();
+                IUntypedColumn utc = new UntypedColumn<ColorColumn.ComparableColor>(coreColumn);
+
+                utc.Name = details.Name;
+
+                return utc;
+            });
+        }
+    }
+
+    public class ColorTable : Table
+    {
+        public ColorTable(string name, long requiredItemCount) : base(name, requiredItemCount)
+        {
+        }
+
+        public void BindColorColumns(string sourceColumn, string targetColorColumn)
+        {
+            foreach (Partition p in GetPartitions())
+            {
+                IUntypedColumn srcColumn = p.Columns[sourceColumn];
+                IUntypedColumn colorColumn = p.Columns[targetColorColumn];
+                (colorColumn.InnerColumn as ColorColumn).LookupColumn = (IColumn<short>)srcColumn.InnerColumn;
+            }            
+        }
+    }
+
+    public class ColorColumn : IColumn<ColorColumn.ComparableColor>
```

Comment:
It'd be awesome if you could inherit ValueTypeColumn<short> to hold the values, and then override this[]. I guess we'd need to look through ValueTypeColumn and make members virtual to make this clean, and we have to watch for performance impact.